### ISSUE TITLE
`semi: never` not working with `do` expressions

### ISSUE THREAD

**Issue by hacker-DOM** [babel/babel-eslint#814]
Hi! I have options such as:
```js
"semi": ["error", "never", {
  "beforeStatementContinuationChars": "always"
}],
```
and code such as

```jsx
const View = (props) => do {
  const {className} = props;
  <div {...{className}}>
    <h1>Home</h1>
  </div>
}
```

Despite being semantically correct (omitting the semicolon does not compile), I get a linting error for the extra semicolon.

I am using 
- `@babel/plugin-proposal-do-expressions@7.7.4`,
- `eslint@6.8.0`,
- `babel-eslint@11.0.0-beta.2`.

**Comment by kaicataldo** [comment]
Thanks for reporting this! We're in the process of migrating this project to the main babel/babel repository, and I have added this issue to the the [list](https://github.com/babel/babel/issues/10752) of things I need to ensure are working.

**Comment by kaicataldo** [comment]
Thank you for the PR. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
